components/aws/sagemaker/common/sagemaker_component.py

```diff
@@ -0,0 +1,561 @@
+                license_file.readline(),
+            )
+            if version_match is not None:
+                component_version = version_match.group(1)
+
+        return component_version
+
+    @staticmethod
+    def _get_common_path() -> str:
+        """Gets the path of the common directory in the project.
+
+        Returns:
+            str: The `realpath` representation of the common directory.
+        """
+        return os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
+
+    @staticmethod
+    def _get_request_template(template_name: str) -> Dict[str, Any]:
+        """Loads and returns a template file as a Python construct.
+
+        Args:
+            template_name: The name corresponding to the template file to load.
+
+        Returns:
+            dict: A Python construct created by loading the template file.
+        """
+        with open(
+            os.path.join(
+                SageMakerComponent._get_common_path(),
+                "templates",
+                f"{template_name}.template.yaml",
+            ),
+            "r",
+        ) as f:
+            request = yaml.safe_load(f)
+        return request
+
+    def _print_log_header(self, header_len, title=""):
+        """Prints a header section for logs.
+
+        Args:
+            header_len: The maximum length of the header line.
+            title: The header title.
+        """
+        logging.info(f"{title:*^{header_len}}")
+
+    def _print_cloudwatch_logs(self, log_grp: str, job_name: str):
+        """Gets the CloudWatch logs for SageMaker jobs.
+
+        Args:
+            log_grp: The name of a CloudWatch log group.
+            job_name: The name of the job as defined in CloudWatch.
+        """
+
+        CW_ERROR_MESSAGE = "Error in fetching CloudWatch logs for SageMaker job"
+
+        try:
+            logging.info(
+                "\n******************** CloudWatch logs for {} {} ********************\n".format(
+                    log_grp, job_name
+                )
+            )
+
+            log_streams = self._cw_client.describe_log_streams(
+                logGroupName=log_grp, logStreamNamePrefix=job_name + "/"
+            )["logStreams"]
+
+            for log_stream in log_streams:
+                logging.info("\n***** {} *****\n".format(log_stream["logStreamName"]))
+                response = self._cw_client.get_log_events(
+                    logGroupName=log_grp, logStreamName=log_stream["logStreamName"]
+                )
+                for event in response["events"]:
+                    logging.info(event["message"])
+
+            logging.info(
+                "\n******************** End of CloudWatch logs for {} {} ********************\n".format(
+                    log_grp, job_name
+                )
+            )
+        except Exception as e:
+            logging.error(CW_ERROR_MESSAGE)
+            logging.error(e)
+
+    def _get_model_artifacts_from_job(self, job_name: str):
```

Comment:
is there a better way to place these methods ?

I think maybe a file where built_in_algos was kept and hyperparameter tuning component can also import from there. 

because there can be shared but component specific methods in future as well

Reply:
Yes it might be helpful for both of them to inherent from some other class that allows for these methods. For the sake of simplicity in this PR (which I've probably already gone well over) I think I might just leave these here for now?